src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1562,13 +1562,18 @@ def process_remove_identity_namespace(cmd, namespace):
 
 
 def process_gallery_image_version_namespace(cmd, namespace):
-    TargetRegion = cmd.get_models('TargetRegion')
+    TargetRegion, EncryptionImages, OSDiskImageEncryption, DataDiskImageEncryption = cmd.get_models(
+        'TargetRegion', 'EncryptionImages', 'OSDiskImageEncryption', 'DataDiskImageEncryption')
     storage_account_types_list = [item.lower() for item in ['Standard_LRS', 'Standard_ZRS', 'Premium_LRS']]
     storage_account_types_str = ", ".join(storage_account_types_list)
 
     if namespace.target_regions:
+        if hasattr(namespace, 'target_region_encryption') and namespace.target_region_encryption:
+            if len(namespace.target_regions) != len(namespace.target_region_encryption):
+                raise CLIError(
+                    'usage error: Length of --target-region-encryption should be as same as length of target regions')
         regions_info = []
-        for t in namespace.target_regions:
+        for i, t in enumerate(namespace.target_regions):
             parts = t.split('=', 2)
```

Comment:
not related to this PR directly, but it's seldom to use = as seperator while meaning is not equal

Reply:
Yes. It is designed by a former CLI member.